cg/store/api/find_business_data.py

```diff
@@ -330,13 +344,94 @@ def get_number_of_reads_for_sample_passing_q30_threshold(
         reads_count: Optional[int] = total_reads_query.scalar()
         return reads_count if reads_count else 0
 
+    def get_number_of_reads_for_sample_on_flow_cell_from_sample_lane_metrics(
+        self, sample_internal_id: str, flow_cell_name: str
+    ) -> int:
+        total_reads_query: Query = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=[
+                SequencingMetricsFilter.FILTER_BY_FLOW_CELL_NAME,
+                SequencingMetricsFilter.FILTER_TOTAL_READ_COUNT_FOR_SAMPLE,
+            ],
+            sample_internal_id=sample_internal_id,
+            flow_cell_name=flow_cell_name,
+        )
+        reads_count: Optional[int] = total_reads_query.scalar()
+        return reads_count if reads_count else 0
+
+    def get_samples_internal_ids_on_flow_cell_from_sample_lane_metrics(
+        self, flow_cell_name: str
+    ) -> List[str]:
+        """Get samples on a given flow cell."""
+        metrics = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=[SequencingMetricsFilter.FILTER_BY_FLOW_CELL_NAME],
+            flow_cell_name=flow_cell_name,
+        ).all()
+
+        return list(set([metric.sample_internal_id for metric in metrics]))
+
+    def get_average_passing_q30_for_sample_from_sample_lane_metrics(
+        self, sample_internal_id: str, flow_cell_name: str
+    ) -> float:
+        """Get average q30 for a sample on a given flow cell."""
+        sample_lane_metrics_filter_functions = [
+            SequencingMetricsFilter.FILTER_BY_FLOW_CELL_NAME,
+            SequencingMetricsFilter.FILTER_BY_SAMPLE_INTERNAL_ID,
+        ]
+        metrics = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=sample_lane_metrics_filter_functions,
+            sample_internal_id=sample_internal_id,
+            flow_cell_name=flow_cell_name,
+        ).all()
+        total_passing_q30: float = 0
+        for metric in metrics:
+            total_passing_q30 += metric.sample_base_fraction_passing_q30
```

Comment:
I feel like this should be weighted by the number of reads on the lane, to be accurate

Reply:
It is a percentage of bases that passes a filter. We sum the percentages here and then get the average when we return

Reply:
Yes, but should the average not be weighted? If we have two lanes one with 10 reads and q30 of 50 and the other lane has 1000 reads and a q30 of 90, then the average should not be 70 right? Rather something like (10*50+1000*90)/(10+1000)

Reply:
it is a measure on base level not reads, so you would want normalise to total bases per lane?

Reply:
Does that not make sense? I feel like the number of bases will always be proportional to the number of reads. And if we get much more data from one lane then the quality of that lane should weigh more than the results from other lanes

Reply:
just a scenario I could think of:
lane 1 low reads, abmysmal q30.
lane 2 high reads, good q30 
Weighted average passes QC threshold which causes poor data to be delivered. 

I think this suggestion would change the current implementation (taken an average) as a cutoff. Maybe we can make an issue and discuss this in a refinement meeting?

Reply:
Issue created: https://github.com/Clinical-Genomics/cg/issues/2285